docs/csharp/language-reference/builtin-types/nullable-reference-types.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Nullable reference types - C# reference"
+description: Learn about C# nullable reference types and how to use them
+ms.date: 04/06/2020
+---
+# Nullable reference types (C# reference)
+
+> [!NOTE]
+> This article covers nullable reference types. You can also declare [nullable value types](nullable-value-types.md).
+
+Nullable reference types are available beginning with C# 8.0, in code that has opted in to a *nullable aware context*. Nullable reference types, the null static analysis warnings, and the [null-forgiving operator](../operators/null-forgiving.md) are optional language features. All are turned off by default. A *nullable context* is controlled at the project level using build settings, or in code using pragmas.
+
+ In a nullable aware context:
+
+- A variable of a reference type `T` must be initialized with non-null, and may never be assigned a value that may be `null`.
+- A variable of a reference type `T?` may be initialized with `null` or assigned `null`, but is required to be checked against `null` before de-referencing.
+- A variable `m` of type `T?` is considered to be non-null when you apply the null-forgiving operator, as in `m!`.
+
+The distinctions between a non-nullable reference type `T` and a nullable reference type `T?` are performed by the compiler's interpretation of the preceding rules. A variable of type `T` and a variable of type `T?` are represented by the same .NET type. The following example declares a non-nullable string, a nullable string, and uses the null-forgiving operator to assign a value to a non-nullable string:
```

Comment:
I thought the original wording that was replaced by "the same .NET type" was clearer.

```suggestion
The distinctions between a non-nullable reference type `T` and a nullable reference type `T?` are enforced by the compiler's interpretation of the preceding rules. A variable of type `T` and a variable of type `T?` are represented by the same type in IL and at run time. The following example declares a non-nullable string and a nullable string, and then uses the null-forgiving operator to assign a value to a non-nullable string:
```

Reply:
To avoid the conflicts, I added those suggestions by hand.